src/nvim/ex_cmds.c

```diff
@@ -5846,3 +5912,299 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
+    // don't use a new tab page
+    cmdmod.tab = 0;
+
+    // Create a window for the command-line buffer.
+    if (win_split((int)p_cwh, WSP_BOT) == FAIL) {
+      beep_flush();
+      unblock_autocmds();
+      return;
+    }
+    cmdwin_type = get_cmdline_type();
+
+    // Create the command-line buffer empty.
+    (void)do_ecmd(0, NULL, NULL, NULL, ECMD_ONE, ECMD_HIDE, NULL);
+    (void)setfname(curbuf, (char_u *) "[live_sub]", NULL, true);
+    set_option_value((char_u *) "bt", 0L, (char_u *) "nofile", OPT_LOCAL);
+    set_option_value((char_u *) "swf", 0L, NULL, OPT_LOCAL);
+    curbuf->b_p_ma = false;  // Not Modifiable
+    curwin->w_p_fen = false;
+    curwin->w_p_rl = cmdmsg_rl;
+    cmdmsg_rl = false;
+    RESET_BINDING(curwin);
+
+    // Do execute autocommands for setting the filetype (load syntax).
+    unblock_autocmds();
+
+    // Showing the prompt may have set need_wait_return, reset it.
+    need_wait_return = false;
+
+    // Reset 'textwidth' after setting 'filetype'
+    // (the Vim filetype plugin sets 'textwidth' to 78).
+    curbuf->b_p_tw = 0;
+
+    // Save the buffer used in the split
+    livebuf = curbuf;
+
+    // Initialize line and highlight variables
+    int line = 0;
+    int src_id_highlight = 0;
+    long sub_size = STRLEN(sub);
+    long pat_size = STRLEN(pat);
+
+    // Get the width of the column which display the number of the line
+    linenr_T highest_num_line = 0;
+    kl_iter(matchedline_T, lmatch, current) {
+      highest_num_line = (*current)->data.lnum;
+    }
+
+    // computing the length of the column that will display line number
+    //TODO: Is log10 efficient for counting the nr of digits?
+    int col_width = log10(highest_num_line) + 1 + 3;
+    // better? log10 might not be precise for large numbers (ref?)
+    //int col_width = snprintf(NULL, 0, "%ld", highest_num_line) + 3;
+
+    // allocate a line sized for the window
+    char *str = xcalloc((size_t)curwin->w_frame->fr_width, sizeof(char));
+
+    // Append the lines to our buffer
+    kl_iter(matchedline_T, lmatch, current) {
+      matchedline_T mat = (*current)->data;
+      size_t line_size = STRLEN(mat.line) + col_width + 1;
+
+      // Reallocation if str not long enough
+      if (line_size > curwin->w_frame->fr_width * sizeof(char)) {
+        str = xrealloc(str, line_size * sizeof(char));
+      }
+
+      // put ' [ lnum]line' into str and append it to the livesub buffer
+      str[0] = ' ';
+      str[1] = '[';
+
+      int i = 2;
+      for ( ; i < col_width-(log10(mat.lnum)+1) - 1; i++) { str[i] = ' '; }
+
+      snprintf(str+i, line_size - i, "%ld%s%s", mat.lnum, "]", mat.line);
+      ml_append(line++, (char_u *)str, (colnr_T)0, false);
+
+      i = 0;
+      kl_iter(colnr_T, mat.start_col, col) {
+        // highlight the replaced part
+        if (sub_size > 0) {
+          src_id_highlight =
+          bufhl_add_hl(curbuf,
+                       src_id_highlight,
+                       2,  // id of our highlight
```

Comment:
Ok, I think I got it :) Care to take a [look](https://github.com/neovim/neovim/pull/4915/commits/01e52d03b57c4b02d7d47732d527557d37112b10)?